doc/credentials.md

```diff
@@ -0,0 +1,223 @@
+#   credentials002 = "11:22:33:44:55:66,ff:ee:dd:cc:bb:aa,ab:12:12:34:34:56:56"
+#
+# The special group 'NoAuth' defines mac addresses of cameras where no authentication is needed.
+# The 'NoAuth' group does not exist in the SecretStore. It is not required to place cameras in here,
+# however it avoids sending the default credentials to cameras which do not need it.
+#
+# IMPORTANT: A MAC Address may only exist in one credential group. If a MAC address is defined in more
+# than one group, it is random which group the MAC will end up in! If you wish to change the group a MAC
+# address belongs to, first remove it from its existing group, and then add it to the new one.
+[AppCustom.CredentialsMap]
+NoAuth = "44:33:22:11:33:fd"
+credentials001 = "aa:bb:cc:dd:ee:ff"
+credentials002 = "11:22:33:44:55:66,ff:ee:dd:cc:bb:aa,ab:12:12:34:34:56:56"
+```
+
+## Credential Lookup
+Here is an in-depth look at the logic behind mapping `Credentials` to Devices.
+
+### During Discovery
+```mermaid
+%% Note: The node and edge definitions are split up to make it easier to adjust the
+%% links between the various nodes.
+graph TD;   
+    %% -------- Node Definitions -------- %%
+    discover[Discovered Device]
+    default[Use Default Credentials]
+    refmac{Does EndpointRef<br/>contain<br/>MAC Address?}
+    noauth{MAC Belongs<br/>to NoAuth group?}
+    none[Set AuthMode to 'none']
+    apply[Apply Credentials]
+    instore{Credentials exist<br/>in SecretStore?}
+    info[Get Device Information]
+    success{Success?}
+    net[Get Network Interfaces]
+    create[Create Device:<br/>&ltMfg&gt-&ltModel&gt-&ltEndpointRef&gt]
+    unk[Create Device:<br/>unknown_unknown_&ltEndpointRef&gt]
+
+    %% -------- Graph Definitions -------- %%
+    discover-->refmac
+    subgraph For all MAC Addresses<br/>in CredentialsMap
+      refmac
+    end
+    refmac--Yes-->noauth
+    refmac--NoMatches-->default
+    noauth--Yes-->none
+    noauth--No-->instore
+    default-->instore
+    none-->info
+    instore--Yes-->apply
+    instore--No-->none
+    apply-->info
+    info--Failed-->unk
+    info--Success-->net
+    net---->create
+```
+
+### Connecting to Existing Devices
+```mermaid
+%% Note: The node and edge definitions are split up to make it easier to adjust the
+%% links between the various nodes.
+graph TD;
+    %% -------- Node Definitions -------- %%
+    exist[Existing Device]
+    hasmac{Device Metadata contains<br/>MAC Address?}
+    map{MAC exists in<br/>CredentialsMap?}
+    noauth{MAC Belongs<br/>to NoAuth group?}
+    default[Use Default Credentials]
+    none[Set AuthMode to 'none']
+    apply[Apply Credentials]
+    instore{Credentials exist<br/>in SecretStore?}
+
+    %% -------- Graph Definitions -------- %%
+    exist-->hasmac
+    hasmac--Yes-->map
+    map--Yes-->noauth
+    hasmac--No-->default
+    noauth--Yes-->none
+    noauth--No-->instore
+    default-->instore
+    instore--Yes-->apply
+    instore--No-->none
+```
```

Comment:
Really like the addition of these flow charts. Very clear on how registration and credential lookup. Really nice!

Reply:
I'll second that, they are a great addition!